### PR TITLE
making storage profile truly optional fixes #39

### LINKED ISSUE #39: Cannot find the VDC storage policy with v0.10
Hi,

After updating 0.9.6 to v0.10, I can no longer create VMs on vCloud
```
* vcd_vapp.terravm: Error creating vapp: &errors.errorString{s:"Error creating vapp: &errors.errorString{s:\"error instantiating a new vApp: API Error: 400: [ e172bcb4-db68-49b4-bd87-8d8d04a95adb ] Cannot find the VDC storage policy \\\"545fb858-3a90-4801-86d4-00d275ae1b6d\\\".\"}"}
```

Switching back to terraform 0.9.6, the same main.tf works fine.

There is no storage reference in the TF file, and its not complicated:
```
provider "vcd" {
  user                 = "${var.username}"
  password             = "${var.password}"
  org                  = "${var.org}"
  url                  = "${var.url}"
  vdc                  = "${var.vdc}"
  maxRetryTimeout      = "${var.maxRetryTimeout}"
  allow_unverified_ssl = "${var.allow_unverified_ssl}"
}
resource "vcd_vapp" "terravm" {
  name          = "${var.vmname}"
  catalog_name  = "${var.catalog}"
  template_name = "${var.template}"
  memory        = "${var.memory}"
  cpus          = "${var.cpus}"
  metadata {
    owner       = "${var.owner}"
    description = "${var.description}"
  }
  network_name = "${var.network}"
  ip = "${var.ip}"
..
```
Any ideas?

### PR DESCRIPTION
**Pull request by bobbydeveaux**
making storage profile truly optional fixes #39

### CODE REVIEW

File: vcd/resource_vcd_vapp_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func TestAccVcdVApp_PowerOff(t *testing.T) {
 			},
 
 			resource.TestStep{
-				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY")),
+				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY"), os.Getenv("VCD_EDGE_GATEWAY")),
```
Comment (tombuildsstuff) [inline vcd/resource_vcd_vapp_test.go]: I think this constant needs to be updated to support the third variable?

File: vcd/resource_vcd_vapp_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ func TestAccVcdVApp_PowerOff(t *testing.T) {
 		CheckDestroy: testAccCheckVcdVAppDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY")),
+				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY"), os.Getenv("VCD_EDGE_GATEWAY")),
```
Comment (tombuildsstuff) [inline vcd/resource_vcd_vapp_test.go]: same here

File: vcd/resource_vcd_vapp.go [hunk 3]
```diff
@@ -117,25 +117,12 @@ func resourceVcdVAppCreate(d *schema.ResourceData, meta interface{}) error {
...
 			if d.Get("storage_profile").(string) != "" {
 				storage_profile_reference, err = vcdClient.OrgVdc.FindStorageProfileReference(d.Get("storage_profile").(string))
 				if err != nil {
-					storage_profile_reference = default_storage_profile
-					log.Printf("Using default storage_profile: %s", storage_profile_reference)
+					return fmt.Errorf("Error finding storage profile %s", d.Get("storage_profile").(string))
 				}
-			} else {
-				storage_profile_reference = default_storage_profile
 			}
```
Comment (tombuildsstuff) [inline vcd/resource_vcd_vapp.go]: 🤔 the alternative here would be to make `storage_profile` a Computed field?

**Comment by bobbydeveaux** [comment]
Status | Test | Duration | Order#
-- | -- | -- | --
OK | TestAccVcdVApp_PowerOff | 10m:22s,846ms | 8
OK | TestAccVcdVAppVm_Basic | 7m:58s,006ms | 9
OK | TestAccVcdVAppRaw_Basic | 4m:08s,109ms | 7
OK | TestAccVcdNetwork_Basic | 1m:23s,398ms | 5
OK | TestAccVcdFirewallRules_basic | 1m:02s,175ms | 4
OK | TestAccVcdVpn_Basic | 57s,560ms | 3
OK | TestAccVcdDNAT_Basic | 55s,444ms | 1
OK | TestAccVcdDNAT_tlate | 55s,170ms | 2
OK | TestAccVcdSNAT_Basic | 54s,809ms | 6

**Pull request merged**